### PR TITLE
Add practice exercise `robot-simulator`

### PR DESCRIPTION
**Pull request by jiegillet**
A nice simple one to get used to pattern matching.

### CODE REVIEW

File: exercises/practice/robot-simulator/.meta/example.gleam [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  West
+}
+
+pub type Position {
+  Position(x: Int, y: Int)
+}
+
+pub fn create(
+  direction direction: Direction,
+  position position: Position,
```
Comment (lpil) [inline exercises/practice/robot-simulator/.meta/example.gleam:21]: Is there a reason we've gone for labels with this one? I feel the tests would be a bit easier to read without them as the two arguments are clearly distinct so the labels are more noise than a help.

**Review by lpil — APPROVED** [review]
Very nice!

File: exercises/practice/robot-simulator/.meta/example.gleam (hunk 1 above)
Comment (jiegillet) [reply exercises/practice/robot-simulator/.meta/example.gleam:21]: Sounds good. I'm sometimes on the fence when there are two arguments, but you are right, the arguments are really distinct here.

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**